### PR TITLE
completed week 3 training, adding comments on my portfolio …

### PR DESCRIPTION
**Pull request by trinitydonohugh**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package com.google.sps.data;
+
+/** A Comment on the comments page. */
+public final class Comment {
+
+  private final long id;
+  private final String name;
+  private final String message;
+  private final String date;
+  //private final long timestamp;
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:24]: Unused code, did you mean to remove them?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  public Comment(long id, String name, String message, String date) {
+    this.id = id;
+    this.name = name;
+    this.message = message;
+    this.date = date;
+    //this.timestamp = timestamp;
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:31]: Ditto.

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  }
+}
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:33]: nit: no newline at EOF

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,42 +14,90 @@
...
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<Comment>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      String message = (String) entity.getProperty("message");
+      String date = (String) entity.getProperty("date");
+      //long timestamp = (long) entity.getProperty("timestamp");
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:92]: Ditto.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -14,42 +14,90 @@
...
+    long timestamp = System.currentTimeMillis();
+
+    // Convert timestamp into human-readable date
+    SimpleDateFormat formatter= new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
+    String date = formatter.format(timestamp);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("message", message);
+    commentEntity.setProperty("date", date);
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:56]: I wonder if you can do this timestamp -> date conversion later. It looks like you don't need both date and timestamp in the data store.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
+
+    Key CommentEntityKey = KeyFactory.createKey("Comment", id);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.delete(CommentEntityKey);
+  }
+}
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:39]: nit: no newline at EOF

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -34,18 +34,48 @@ function addRandomGreeting() {
  * Fetches comments from the servers and adds them to the DOM.
  */
 function getMessages() {
-  fetch('/data').then(response => response.json()).then((messages) => {
-    const messagesListElement = document.getElementById('messages-container');
-    messagesListElement.innerHTML = 'messages: ';
-    for (i = 0; i < messages.length; i++) {
-      messagesListElement.appendChild(createListElement(messages[i]));
-    }
+  //document.getElementById('messages-container').innerHTML = "";
```
Comment (pengxianzhe) [inline portfolio/src/main/webapp/script.js:37]: Unused code, did you mean to remove them?

**Comment by trinitydonohugh** [comment]
https://screenshot.googleplex.com/Hd2rcyKUmPc.png -- link with details on style changes to website.

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 1 above)
Comment (trinitydonohugh) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:24]: Yes -- so sorry again! I will make sure to make those changes now and submit a new pull request.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (trinitydonohugh) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:56]: I just tried it and I was able to do the timestamp -> date conversion in the doGet section! I'll make final touches to that now and commit the change. Thanks for the tip!

**Pull request merged**